staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -58,6 +66,20 @@ func getEventKey(event *v1.Event) string {
 		"")
 }
 
+// getSpamKey builds unique event key based on source, involvedObject
+func getSpamKey(event *v1.Event) string {
+	return strings.Join([]string{
```

Comment:
It's more of a "go-ism" thing - "don't use strings when you can use structs".  Strings can have false sharing unless you have a delimiter (this can be used to maliciously rate limit someone else's events).